third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -1120,6 +1188,100 @@ index 11664f66c..ee37ddd2d 100755
          context_layer = context_layer.view(new_context_layer_shape)
  
          outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+diff --git a/src/transformers/optimization.py b/src/transformers/optimization.py
+index b957acb6d..8fb24dd9c 100644
+--- a/src/transformers/optimization.py
++++ b/src/transformers/optimization.py
+@@ -22,6 +22,7 @@ import torch
+ from torch import nn
+ from torch.optim import Optimizer
+ from torch.optim.lr_scheduler import LambdaLR
++from nncf.common.schedulers import BaseCompressionScheduler
+ 
+ from .trainer_utils import SchedulerType
+ from .utils import logging
+@@ -31,6 +32,18 @@ from .utils.versions import require_version
+ logger = logging.get_logger(__name__)
+ 
+ 
++class BNASLambdaLR(LambdaLR, BaseCompressionScheduler):
++    def __init__(self, optimizer, lr_lambda, last_epoch=-1, verbose=False):
++        LambdaLR.__init__(self, optimizer, lr_lambda, last_epoch, verbose)
++        BaseCompressionScheduler.__init__(self)
++
++    def epoch_step(self, next_epoch = None):
++        pass
++
++    def stage_step(self, stage_desc):
++        pass
++
++
+ def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
+     """
+     Create a schedule with a constant learning rate, using the learning rate set in optimizer.
+@@ -44,7 +57,7 @@ def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
+     Return:
+         `torch.optim.lr_scheduler.LambdaLR` with the appropriate schedule.
+     """
+-    return LambdaLR(optimizer, lambda _: 1, last_epoch=last_epoch)
++    return BNASLambdaLR(optimizer, lambda _: 1, last_epoch=last_epoch)
```

Comment:
I replaced all LambdaLR with BNASLambdaLR. In this case, we can use all lr schedulers available in HuggingFace.